core-rust/core-api-server/src/core_api/handlers/lts/transaction_status.rs

```diff
@@ -30,9 +30,7 @@ pub(crate) async fn handle_lts_transaction_status(
         pending_transaction_result_cache.peek_all_known_payloads_for_intent(&intent_hash);
     drop(pending_transaction_result_cache);
 
-    // TODO(locks): consider this (and other) usages of DB "read current" lock. *Carefully* migrate
```

Comment:
Hmh so we're now allowing (locking-wise) concurrent writes to VertexStoreCf (in `save_vertex_store` and `commit`).
While I don't think it can cause issues at the moment given the context in which those methods are used (e.g. we only include vertex store in `commit` if it originates from consensus, and that's the same thread that calls `save_vertex_store`), I'd still suggest to change that to `lock` just in case (e.g. if some assumptions change in the future). In normal operating conditions this lock will always be non-contentious.

Reply:
yeah, let me add the lock, since it indeed writes to the same "db region" as `commit` 👍